### PR TITLE
[CARBONDATA-32] Refactor DataType related code

### PR DESCRIPTION
**Pull request by jackylk**
1. Merge two DataType
2. Merge tow DataTypeUtil
3. Remove DataSetUtil

### CODE REVIEW

File: core/src/main/java/org/carbondata/core/util/DataTypeUtil.java [hunk 1]
```diff
@@ -1,3 +1,22 @@
+/*
```
Comment (ravipesala) [inline core/src/main/java/org/carbondata/core/util/DataTypeUtil.java]: It seems the latest code of `org.carbondata.scan.util.DataTypeUtil` is not merged properly. Please take the latest code of `DataTypeUtil` and merge it.

File: core/src/main/java/org/carbondata/scan/filter/executer/RowLevelFilterExecuterImpl.java [hunk 2]
```diff
@@ -178,7 +177,7 @@ private void createRow(BlocksChunkHolder blockChunkHolder, RowIntf row, int inde
                 memberString = null;
               }
             }
-            record[dimColumnEvaluatorInfo.getRowIndex()] = DataTypeUtil
+            record[dimColumnEvaluatorInfo.getRowIndex()] = org.carbondata.core.util.DataTypeUtil
```
Comment (ravipesala) [inline core/src/main/java/org/carbondata/scan/filter/executer/RowLevelFilterExecuterImpl.java]: Better import the package in all places

File: core/src/main/java/org/carbondata/scan/result/preparator/impl/RawQueryResultPreparatorImpl.java [hunk 3]
```diff
@@ -92,7 +91,7 @@ public RawQueryResultPreparatorImpl(QueryExecutorProperties executerProperties,
           int noDictionaryColumnIndex = 0;
           for (int i = 0; i < dimSize; i++) {
             if (!queryDimensions[i].getDimension().hasEncoding(Encoding.DICTIONARY)) {
-              row[order[i]] = DataTypeUtil.getDataBasedOnDataType(
+              row[order[i]] = org.carbondata.core.util.DataTypeUtil.getDataBasedOnDataType(
```
Comment (ravipesala) [inline core/src/main/java/org/carbondata/scan/result/preparator/impl/RawQueryResultPreparatorImpl.java]: import the package

File: integration/spark/src/main/scala/org/carbondata/spark/util/CarbonScalaUtil.scala [hunk 4]
```diff
@@ -67,16 +65,15 @@ object CarbonScalaUtil {
...
-      case DataType.STRING => StringType
-      case DataType.SHORT => ShortType
-      case DataType.INT => IntegerType
-      case DataType.LONG => LongType
-      case DataType.DOUBLE => DoubleType
-      case DataType.BOOLEAN => BooleanType
-      case DataType.DECIMAL => DecimalType.SYSTEM_DEFAULT
-      case DataType.TIMESTAMP => TimestampType
+      case CarbonDataType.STRING => StringType
+      case CarbonDataType.INT => IntegerType
```
Comment (ravipesala) [inline integration/spark/src/main/scala/org/carbondata/spark/util/CarbonScalaUtil.scala]: `ShortType` is missing

File: integration/spark/src/main/scala/org/carbondata/spark/util/CarbonScalaUtil.scala [hunk 5]
```diff
@@ -24,29 +24,27 @@ import org.apache.spark.sql.execution.command.Level
...
-      case FloatType => CarbonDataType.FloatType
-      case DateType => CarbonDataType.DateType
-      case BooleanType => CarbonDataType.BooleanType
-      case TimestampType => CarbonDataType.TimestampType
-      case ArrayType(_, _) => CarbonDataType.ArrayType
-      case StructType(_) => CarbonDataType.StructType
-      case NullType => CarbonDataType.NullType
-      case _ => CarbonDataType.DecimalType
+      case StringType => CarbonDataType.STRING
+      case IntegerType => CarbonDataType.INT
```
Comment (ravipesala) [inline integration/spark/src/main/scala/org/carbondata/spark/util/CarbonScalaUtil.scala]: `ShortType` is missing

**Comment by ravipesala** [comment]
@jackylk Some test cases are failing on this PR, it may be because of latest `DataTypeUtil` is not merged. Please update it run the test cases.

File: core/src/main/java/org/carbondata/core/util/DataTypeUtil.java (hunk 1 above)
Comment (jackylk) [inline core/src/main/java/org/carbondata/core/util/DataTypeUtil.java]: yes, it is done

**Comment by jackylk** [comment]
retest this please

**Comment by ravipesala** [comment]
LGTM

**Comment by ravipesala** [comment]
@chenliang613 Please merge this PR

**Pull request closed without merging**